### PR TITLE
Added automated release process to project pipeline

### PR DESCRIPTION
**Pull request by neil-k-zero**
- Reconfigured bin/publish to perform releases and promotions
of contianer images based on optional flag parameters.

- Added additional functions in bin/build_utils for the automated releases.

- Maintained legacy bin/build_utils functions that are currently no longer in
use, but might provide useful in the future.

- Added submodule for cyberark/conjur-opentelemetry-tracer upstream dependency.

- Added submodule for cyberark/conjur-authn-k8s-client upstream dependency.

- Added submodule for cyberark/conjur-api-go upstream dependency.

- During a build, images get pushed inertnally to registry.tld. These
images allow for promotion at a later time if/when necessary.

### Desired Outcome

Add project to automated release process.

### Implemented Changes

- Version file is automatically generated from the CHANGELOG, this is utilized for automated
releases.

- Github pre-release are automatically generated with the option to promote said pre-release
to a GitHub release for customers.

- Docker Software Bill of Materials is generated on releases and is stored as an artifact. This can
be utilized to regenerate the go.mod file.

- Builds will tag and push images internally to registry.tld.

- Container images are tagged as edge for pre-releases.

- Images that are promoted (released) are tagged as latest. 

### Connected Issue/Story

N/A

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Comment by jtuttle** [comment]
This looks like great, are you planning on updating the [release instructions](https://github.com/cyberark/secrets-provider-for-k8s/blob/main/CONTRIBUTING.md#releases) to match the new workflow?

**Comment by neil-k-zero** [comment]
> This looks like great, are you planning on updating the [release instructions](https://github.com/cyberark/secrets-provider-for-k8s/blob/main/CONTRIBUTING.md#releases) to match the new workflow?

Ah, sure thing, I'll work on updating it.

**Comment by jtuttle** [comment]
Thanks, we'll want to make sure to do update release instructions on any repos that undergo this change so that we don't end up with someone trying to release the old way.

**Comment by neil-k-zero** [comment]
> Thanks, we'll want to make sure to do update release instructions on any repos that undergo this change so that we don't end up with someone trying to release the old way.

@jtuttle I noticed there's a section in the release doc that has a few helm chart yaml files that need to be manually updated. I'm thinking I could possibly write something that will utilize the now auto generated VERSION file and update those files automatically to match. Then I can remove that entire section from the doc.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+## [1.4.1] - 2022-04-01
+### Changed
+- Update to automated release process
```
Comment (rpothier) [inline CHANGELOG.md:11]: Can you include a link to this PR?

Comment (neil-k-zero) [reply CHANGELOG.md]: Link has been added.

**Review by szh — APPROVED** [review]
LGTM! Just needs a rebase.

**Pull request merged**